share/getters/shrex.go

```diff
@@ -71,10 +73,9 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 		peer, setStatus, err := sg.peerManager.Peer(ctx, root.Hash())
 		if err != nil {
 			log.Debugw("couldn't find peer", "datahash", root.String(), "err", err)
-			return nil, fmt.Errorf("getter/shrex: %w", err)
 		}
 
-		reqCtx, cancel := ctxWithSplitTimeout(ctx, sg.minAttemptsCount, sg.minRequestTimeout)
+		reqCtx, cancel := ctxWithSplitTimeout(ctx, sg.minAttemptsCount-attempt+1, sg.minRequestTimeout)
```

Comment:
So on each new attempt, we decrease the split factor, meaning timeout grows on each attempt until the split factor is zero, correct?